### PR TITLE
Fix authentication in https case

### PR DESCRIPTION
**Pull request by pashaosipyants**
Not sure this fix is correct, didn't dive deep into the code. 
So this PR needs **very** attentive review. 
With this PR, mainly, I wanted to demonstrate the issue #926

### CODE REVIEW

File: core/proxy.go [hunk 1]
```diff
@@ -169,6 +169,9 @@ func unauthorizedError(request *http.Request, realm, message string) *http.Respo
 func proxyBasicAndBearer(proxy *goproxy.ProxyHttpServer, realm string, basicFunc func(user, passwd string) bool, bearerFunc func(token string) bool) {
 
 	proxy.OnRequest().Do(goproxy.FuncReqHandler(func(req *http.Request, ctx *goproxy.ProxyCtx) (*http.Request, *http.Response) {
+		if strings.HasSuffix(req.URL.Host, ":443") {
```
Comment (pashaosipyants) [inline core/proxy.go:172]: in https case auth headers are in connect request, but not in request to destination

File: core/proxy.go [hunk 2]
```diff
@@ -60,6 +52,14 @@ func NewProxy(hoverfly *Hoverfly) *goproxy.ProxyHttpServer {
 		})
 	}
 
+	proxy.OnRequest(matchesFilter(hoverfly.Cfg.Destination)).
```
Comment (pashaosipyants) [inline core/proxy.go:55]: authentication connect handler was unreachable, only this one was called

File: core/proxy.go (hunk 1 above)
Comment (tommysitu) [reply core/proxy.go:172]: that's right, it's not available after the proxy CONNECT

File: core/proxy.go (hunk 2 above)
Comment (tommysitu) [reply core/proxy.go:55]: The handlers are invoked in order it's registered to the proxy, this should fix it. It makes sense to have the auth handler intercept requests first.

**Pull request merged**